### PR TITLE
Deprecate messagestream

### PR DESCRIPTION
**Pull request by surli**
# Jira URL

https://jira.xwiki.org/browse/XWIKI-22438

# Changes

## Description

* Move messagestream module to legacy
* Get rid of messagestream-test module as it was of poor value
* Move translations in the legacy module
* Move notification related templates in legacy module
* Remove calls to the script service in XS code
* Add TODO for handling proper events in wiki manager 

## Clarifications

*

# Screenshots & Video

N/A

# Executed Tests

# Expected merging strategy

* Prefers squash: Yes 
* Backport on branches:
  * No

### CODE REVIEW

File: xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/JoinWiki.xml [hunk 1]
```diff
@@ -105,17 +105,12 @@
...
           ## Accept
           #if ($request.acceptInvitation)
             #set ($discard = $services.wiki.user.acceptInvitation($existingInvitation, $request.userComment))
           #else
             #set ($discard = $services.wiki.user.refuseInvitation($existingInvitation, $request.userComment))
-          #end          
-          ## Send message to wiki admins. (XWikiAdminGroup)
-          #set ($wikiAdminGroupReference = $services.model.createDocumentReference($wikiId, 'XWiki', 'XWikiAdminGroup'))
-          #set ($discard = $services.messageStream.postMessageToGroup($notificationMessage, $wikiAdminGroupReference))
+          #end
```
Comment (michitux) [inline xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/JoinWiki.xml:113]: Isn't removing this code a regression? Or is this somehow added back in the legacy module?

Comment (surli) [reply xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/JoinWiki.xml:113]: See https://jira.xwiki.org/browse/XWIKI-22478

Comment (surli) [reply xwiki-platform-core/xwiki-platform-wiki/xwiki-platform-wiki-ui/xwiki-platform-wiki-ui-mainwiki/src/main/resources/WikiManager/JoinWiki.xml:113]: and also: https://forum.xwiki.org/t/deprecate-and-remove-message-stream-from-xwiki-standard/14780/9?u=surli

TL;DR: the feature is broken for a long time now so removing it isn't causing a regression and we need a proper improvment for it.
